arrays/arrays.c

```diff
@@ -83,14 +107,34 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
+  if (index > arr->count) {
+    switch(errno) {
+      default:
+        fprintf(stderr, "Inserting into array -> Error: Index is out of bounds \n");
+    }
+
+    exit(1);
+  }
 
   // Resize the array if the number of elements is over capacity
+  if (arr->count + 1 > arr->capacity) {
+    resize_array(arr);
+  }
 
   // Move every element after the insert index to the right one position
+  for (int i = index; i < arr->count; i++) {
+    // Incrementing this to i + 2 gives the correct result but failed tests,
+    // but after some debugging and doing `make tests` i + 1 returns
+    // all tests passed, but somehow display incorrect elements in array.
+    arr->elements[i+1] = arr->elements[i];
+  }
 
   // Copy the element and add it to the array
+  char* elem = strdup(element); // Also adding this line here and addign that to the arr->elements[index] causes a segmentation fault.
```

Comment:
It's usually preferred to do char *var_name but it's not strictly necessary unless you're declaring to variables at the same time (e.g. char* a,b) makes it look like both are char*s when only a is in reality.